### ISSUE TITLE
Cannot call confirmSetupIntent with a setup intent created elsewhere

### ISSUE THREAD

**Issue by paulbryan** [stripe/stripe-react-native#325]
**Describe the bug**
Cannot confirm a setupIntent created outside the app

**To Reproduce**
1. Create a setupIntent with payment_method and usage:off_session outside the app
2. use the client_secret to call confirmSetupIntent
3. error returned = {"code":"Failed","message":"You must provide card details"}

**Expected behavior**
The setup intent to be confirmed

**Smartphone:**
 - Device: iphone 12 Pro Max (emulated)
 - OS: iOS14.4

**SDK Verion: 0.1.4**

**Additional context**
sample setup intent
seti_1J0jtQCOnQnnH0At0rVt4Wjg_secret_Je1xyh6q3Wtxpignlr2LltzDf1HvDiC

Invariant Violation: No callback found with cbID 17727 and callID 8863 for  StripeSdk.confirmSetupIntent - most likely the callback was already invoked. Args: '[{"code":"Unknown","message":"Unhandled error occured","domain":"ABI41_0_0RCTErrorDomain","userInfo":null,"nativeStackIOS":["0   Exponent                            0x0000000107919632 ABI41_0_0RCTJSErrorFromCodeMessageAndNSError + 79","1   Exponent                            0x00000001078de01a __50-[ABI41_0_0RCTModuleMethod processMet...(truncated)...","2   Exponent                            0x00000001079e420e $sSo8NSStringCSgACSo7NSErrorCSgIeyByyy_SS...(truncated)...","3   Exponent                            0x00000001079ecf18 $s28ABI41_0_0stripe_react_native9StripeSd...(truncated)...","4   Exponent                            0x00000001079e6749 $s28ABI41_0_0stripe_react_native9StripeSd...(truncated)...","5   CoreFoundation                      0x00000001178d62fc __invoking___ + 140","6   CoreFoundation                      0x00000001178d37b6 -[NSInvocation invoke] + 303","7   CoreFoundation                      0x00000001178d3a47 -[NSInvocation invokeWithTarget:] + 70","8   Exponent                            0x00000001078de824 -[ABI41_0_0RCTModuleMethod invokeWithBrid...(truncated)...","9   Exponent                            0x00000001078e0c72 _ZN17ABI41_0_0facebook14ABI41_0_0ReactL11...(truncated)...","10  Exponent                            0x00000001078e09ed ___ZN17ABI41_0_0facebook14ABI41_0_0React2...(truncated)...","11  libdispatch.dylib                   0x000000011d4f332f _dispatch_call_block_and_release + 12","12  libdispatch.dylib                   0x000000011d4f4508 _dispatch_client_callout + 8","13  libdispatch.dylib                   0x000000011d4fa3f9 _dispatch_lane_serial_drain + 715","14  libdispatch.dylib                   0x000000011d4faf74 _dispatch_lane_invoke + 403","15  libdispatch.dylib                   0x000000011d505577 _dispatch_workloop_worker_thread + 782","16  libsystem_pthread.dylib             0x000000011d9b945d _pthread_wqthread + 314","17  libsystem_pthread.dylib             0x000000011d9b842f start_wqthread + 15"]}]'

**Comment by thorsten-stripe** [comment]
Can you explain your use case, please? Why would you create a SetupIntent if you already have a payment method for the customer? The `confirmSetupIntent` method has to be used in conjunction with the `CardField` component: https://stripe.com/docs/payments/save-and-reuse?platform=react-native

**Comment by paulbryan** [comment]
We want to move to paymentIntents but have some legacy code that is using the old charges api to charge the customers default source (we are upgrading this legacy code but it will take time).  

We tried adding the card info using the stripe cardfield but the legacy code did not charge because it is looking for the default_source and adding the card as a paymentMethod the default is stored in invoice_settings.

What we tried was send the card info to our api, which creates a cardToken, add it to a cardSource, set the customers default_source, then create a setupIntent attach the cardSource and return the setupIntent to the front end if the setupIntent needed action.  What we need is a way to trigger the 3dsecure check on the front end if the setupIntent status is requires_action.
